### PR TITLE
Fixed setting of frame size properties for MSMF-based VideoCapture

### PR DESCRIPTION
**Pull request by terfendail**
### This pullrequest changes
Fix setting of frame size properties for MSMF-based VideoCapture

### CODE REVIEW

File: modules/videoio/src/cap_msmf.cpp [hunk 1]
```diff
@@ -853,31 +857,28 @@ bool CvCapture_MSMF::configureOutput(UINT32 width, UINT32 height, double prefFra
...
-                    ((MT.width * MT.height) > (MTFallback.width * MTFallback.height)) ||
-                    (((MT.width * MT.height) == (MTFallback.width * MTFallback.height)) && getFramerate(MT) > getFramerate(MTFallback) && (prefFramerate == 0 || getFramerate(MT) <= prefFramerate)))
-                {
-                    dwStreamFallback = (int)dwStreamTest;
-                    MTFallback = MT;
-                }
-                if (MT.width == width && MT.height == height)
+                if (dwStreamBest < 0 ||
+                    (
+                        ((width > MT.width ? width - MT.width : width > 0 ? MT.width - width : 0) + (height > MT.height ? height - MT.height : height > 0 ? MT.height - height : 0)) <
```
Comment (vpisarev) [inline modules/videoio/src/cap_msmf.cpp]: the conditions look too complex. Can we simplify them?
e.g.

<pre>
int best_diff = INT_MAX;

for( ... ) {
...
int temp_width = width > 0 ? width : MT.width;
int temp_height = height > 0 ? height : MT.height;
int diff = abs(temp_width - MT.width) + abs(temp_height - MT.heighT);
if( diff < best_diff ) { best_width = MT.width; best_height = MT.height; best_diff = diff; }
}
</pre>

Comment (terfendail) [reply modules/videoio/src/cap_msmf.cpp]: I've refactored the code. Now it should be more readable

File: modules/videoio/src/cap_msmf.cpp [hunk 2]
```diff
@@ -1218,13 +1218,13 @@ double CvCapture_MSMF::getProperty( int property_id ) const
         return aspectN;
     else if (property_id == CV_CAP_PROP_SAR_DEN)
         return aspectD;
+    else if (property_id == CV_CAP_PROP_FRAME_WIDTH)
+        return isOpen ? captureFormat.width : requestedWidth;
```
Comment (alalek) [inline modules/videoio/src/cap_msmf.cpp]: > isOpen ?

How we can create `cv::VideoCapture` where `isOpen == false`? In this case VideoCapture should trying the next backend from the list of enabled/available, and lost "this" instance.

How should we write MSMF test to test both branches of this condition?

**Comment by terfendail** [comment]
Enabled `MODE_HW` by default and simplified set/get logic regarding `isOpen`

**Pull request merged**